table/server/underdb/glue/src/main/java/alluxio/table/under/glue/GlueDatabase.java

```diff
@@ -0,0 +1,367 @@
+    }
+    return DefaultAWSCredentialsProviderChain.getInstance();
+  }
+
+  @Override
+  public String getType() {
+    return GlueDatabaseFactory.TYPE;
+  }
+
+  @Override
+  public String getName() {
+    return mGlueDbName;
+  }
+
+  @Override
+  public List<String> getTableNames() throws IOException {
+    try {
+      String nextToken = null;
+      List<String> tableNames = new ArrayList<>();
+      do {
+        GetTablesRequest tablesRequest =
+            new GetTablesRequest()
+                .withCatalogId(mGlueConfiguration.get(Property.CATALOG_ID))
+                .withDatabaseName(mGlueDbName)
+                .withNextToken(nextToken);
+        GetTablesResult tablesResult = mGlueClient.getTables(tablesRequest);
+        tablesResult.getTableList().forEach(table -> tableNames.add(table.getName()));
+        nextToken = tablesResult.getNextToken();
+      } while (nextToken != null);
+      return tableNames;
+    } catch (EntityNotFoundException e) {
+      throw new IOException("Failed to get glue tables: " + e.getMessage(), e);
+    }
+  }
+
+  private String mountAlluxioPath(String tableName, AlluxioURI ufsUri, AlluxioURI tableUri)
+      throws IOException, AlluxioException {
+    if (Objects.equals(ufsUri.getScheme(), Constants.SCHEME)) {
+      // already an alluxio uri, return the alluxio uri
+      return ufsUri.toString();
+    }
+    try {
+      tableUri = mUdbContext.getFileSystem().reverseResolve(ufsUri);
+      LOG.debug("Trying to mount table {} location {}, but it is already mounted at location {}",
+          tableName, ufsUri, tableUri);
+      return tableUri.getPath();
+    } catch (InvalidPathException e) {
+      // ufs path not mounted, continue
+    }
+    // make sure the parent exists
+    mUdbContext.getFileSystem().createDirectory(tableUri.getParent(),
+        CreateDirectoryPOptions.newBuilder().setRecursive(true).setAllowExists(true).build());
+    Map<String, String> mountOptionMap = mGlueConfiguration.getMountOption(
+        String.format("%s://%s/", ufsUri.getScheme(), ufsUri.getAuthority().toString()));
+    MountPOptions.Builder option = MountPOptions.newBuilder();
+    for (Map.Entry<String, String> entry : mountOptionMap.entrySet()) {
+      if (entry.getKey().equals(UdbConfiguration.READ_ONLY_OPTION)) {
+        option.setReadOnly(Boolean.parseBoolean(entry.getValue()));
+      } else if (entry.getKey().equals(UdbConfiguration.SHARED_OPTION)) {
+        option.setShared(Boolean.parseBoolean(entry.getValue()));
+      } else {
+        option.putProperties(entry.getKey(), entry.getValue());
+      }
+    }
+    mUdbContext.getFileSystem().mount(tableUri, ufsUri, option.build());
+
+    LOG.info("mounted table {} location {} to Alluxio location {} with mountOption {}",
+        tableName, ufsUri, tableUri, option.build());
+    return tableUri.getPath();
+  }
+
+  private PathTranslator mountAlluxioPaths(Table table, List<Partition> partitions)
```

Comment:
similarly , can this be refactored?

Reply:
Wasn't this already moved to a Utils class?

Reply:
Since mountAlluxioPaths required dependency on hive and glue, which is a different method for hive udb and glue udb. So I move mountAlluxioPath to Utils class, and keep mountAlluxioPaths in hive udb and glue udb.